runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -385,6 +385,8 @@ class MM_ObjectAccessBarrierAPI
 				vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
 			} else if (j9gc_modron_readbar_always == _readBarrierType) {
 				vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
+			} else if (gc_modron_readbar_region_check == _readBarrierType) {
+				vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFields(vmThread, objectClass, srcObject, srcOffset, destObject, destOffset);
```

Comment:
This inner 'if' branching was (probably) created early with an idea to eventually implement fast inline copy for range_check barrier, but none has been implement and likely won't be in foreseeable future.
I'd suggest to collapse 3 conditional cases in 1 unconditional call to slow path barrier.

Reply:
I believe I fixed this.  Does this look like what you had in mind?

Reply:
Yes, that's it.

There is similar cleanup you could do for J9MONITORTABLE_OBJECT_LOAD macro.

Reply:
I'll fix that too